src/stacked_borrows.rs

```diff
@@ -467,6 +469,41 @@ impl<'tcx> Stack {
 
         Ok(())
     }
+
+    /// Report a descriptive error when `new` could not be granted from `derived_from`.
+    fn grant_error(
+        &self,
+        derived_from: SbTag,
+        new: Item,
+        alloc_id: AllocId,
+        alloc_range: AllocRange,
+    ) -> InterpError<'static> {
+        let action = format!(
+            "trying to reborrow for {:?} over {:?}[{:#x}..{:#x}]",
+            new.perm,
+            alloc_id,
+            alloc_range.start.bytes(),
+            alloc_range.end().bytes()
+        );
+        if derived_from == SbTag::Untagged {
+            err_sb_ub(format!(
+                "{}, but parent tag {:?} does not grant the required permission over memory range",
```

Comment:
Why does the untagged message not distinguish between the tag existing and not existing?